resources/assets/lib/chat/chat-state-store.ts

```diff
@@ -103,7 +99,9 @@ export default class ChatStateStore implements DispatchListener {
   }
 
   @action
-  selectChannel(channelId: number) {
+  selectChannel(channelId: number, mode: 'advanceHistory' | 'replaceHistory' = 'advanceHistory', updateHistory = true) {
```

Comment:
updateHistory seem redundant if null can be passed for mode (and mode is already a noop when updateHistory is false).